# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsBulletFrame.cpp
# Commit: 5ff1cc0ecdb3
# Full Hash: 5ff1cc0ecdb3cc503d6e0b6f5509eda78d551c7a
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-05-05 04:13:36
# Regressor Bug: 1513447
# File Overlap Count: 1
# Description:
#   Bug 1513447 - Use generated marker elements for list bullet accessibles. r=Jamie
#   
#   In 1513447 there is a demonstrated instance in which the generated
#   marker is replaced with another one and throws the list item bullet
#   state into an unknown state. To remedy this we need to observe when such
# ==============================================================================

diff -r 029a5e8fa8f3 -r 5ff1cc0ecdb3 layout/generic/nsBulletFrame.cpp
--- a/layout/generic/nsBulletFrame.cpp	Tue May 04 18:46:44 2021 +0000
+++ b/layout/generic/nsBulletFrame.cpp	Tue May 04 19:07:59 2021 +0000
@@ -126,9 +126,12 @@
                        !newStyleList->mCounterStyle.IsNone();
 
       if (hadBullet != hasBullet) {
-        nsIContent* listItem = mContent->GetParent();
-        accService->UpdateListBullet(PresContext()->GetPresShell(), listItem,
-                                     hasBullet);
+        if (hadBullet) {
+          accService->ContentRemoved(PresContext()->GetPresShell(), mContent);
+        } else {
+          accService->ContentRangeInserted(PresContext()->GetPresShell(),
+                                           mContent, nullptr);
+        }
       }
     }
   }
